sqlsmith;mutations: blitz to fix Postgres compare_test bugs

rafiss opened the pull request:
See the individual commits. All of the fixes are test modifications, no functional changes.

Release note: None

rafiss committed:
sql/rowenc: use CASE instead of IF() for random computed columns

This allows the randomly create computed column definition to be
used against both CockroachDB and PostgreSQL.

Release note: none

rafiss committed:
compare_test: log name when attempting connection

This prevents an unused variable compilation error.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/62706)
<!-- Reviewable:end -->

rafiss committed:
mutations: allow spatial types for Postgres

Release note: None

rafiss committed:
mutations: prevent context-dependent casts in Postgres

Release note: None

rafiss committed:
mutations: don't allow box2d as a primary key for Postgres

Also changed the interleaving setup so that it no longer tries to interleave
at all, since we are messing around with the PK now.

Release note: None

rafiss committed:
sqlsmith: ignore crdb_internal_mvcc_timestamp in postgres mode

Release note: None

rafiss committed:
sqlsmith: fix date arithmetic in postgres mode

Release note: None

rafiss committed:
sqlsmith: ignore CockroachDB-specific functions in postgres mode

Release note: None

rafiss committed:
sqlsmith: ignore missing binops in postgres mode

Release note: None

rafiss committed:
sqlsmith: use INT4 for Postgres BinOps that require it

Release note: None

rafiss committed:
sqlsmith: use INT4 args for functions in postgres mode

Release note: None

rafiss committed:
sqlsmith: don't ORDER/GROUP BY box2d in postgres mode

Release note: None

rafiss committed:
sqlsmith: fallback to constant args for aggregate functions

It was a bug to fallback to using a non-aggregate function, since it
could result in a column being referenced from HAVING without being in
the GROUP BY or any aggregate functions.

Release note: None

rafiss committed:
mutations: handle tuple formatting difference in postgres

Release note: None

rafiss committed:
testutils: fix inspection of FK constraints

Release note: None

otan reviewed (approved):
ha geez what a slog. thank you for breaking it put into individual commits
i do think you'll find more spatial discrepencies we're ok with though, e.g. st_astext/st_asewkt we have an extra space. wonder how that we didn't get caught o_o

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, 1 of 1 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, 1 of 1 files at r11, 1 of 1 files at r12, 1 of 1 files at r13, 1 of 1 files at r14, 1 of 1 files at r15.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/62706#-:-MXAsKCQ4Em1NMbY3CA7:b-z4fjzm)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @rafiss)



<!-- Sent from Reviewable.io -->

rafiss committed:
compare_test: only use PostgresMutator in postgres test

Release note: None

rafiss commented:
tftr!

bors r=otan
